api/analyzers/source_analyzer.py

```
@@ -34,44 +63,45 @@ def first_pass(self, ignore: List[str], executor: concurrent.futures.Executor) -
+            # Walk thought the AST
+            self.create_hierarchy(analyzer, file)
 
-        # Wait for all tasks to complete
-        #concurrent.futures.wait(tasks)
+            graph.add_file(file)
+            for node, entity in file.entities.items():
+                cls = Class(str(file_path), analyzer.get_entity_name(node), analyzer.get_entity_docstring(node), node.start_point.row, node.end_point.row)
+                graph.add_class(cls)
+                entity.id = cls.id
+                graph.connect_entities("DEFINES", file.id, cls.id)
+                for node, entity in entity.children.items():
```

Ruff (0.8.2)

Loop control variable `entity` overrides iterable it iterates

(B020)